setup.py

```diff
@@ -33,7 +32,9 @@ def read(file_name):
         'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3.5',
     ],
+    install_requires=['boto3', 'six', 'typing'],
```

Comment:
we should not include typing. according to the docs, it's provisional, and may see breaking changes even in minor releases.

Reply:
@jesterhazy, the typing module is only used here for type hinting. Type hinting was introduced by Guido himself in python 3.5. It does not enforce code. I used type hinting here with the objective to be more informative about the functions signatures to our users. The only thing it really does is: 

<img width="443" alt="screen shot 2018-04-16 at 3 55 16 am" src="https://user-images.githubusercontent.com/142074/38814479-6f703af6-4146-11e8-8030-b13d8f472f4f.png">
<img width="442" alt="screen shot 2018-04-16 at 3 57 00 am" src="https://user-images.githubusercontent.com/142074/38814484-7381c4fc-4146-11e8-97d3-b01fd79885e8.png">

You will notice in the codebase that typing is only imported once, and it has being used in the typing comments. 

I agree that we should think about using typing out of the scope of this PR. I will remove it.

Reply:
@jesterhazy I removed the import from typing. I am still using the type hints comments convention, just with basic python types.

That changes the example above to:


![screen shot 2018-04-19 at 6 56 54 am](https://user-images.githubusercontent.com/142074/38996008-125610a6-439f-11e8-8de3-26315d261563.png)

Please, let me know if you are ok with that.